app/check_tx.go

```diff
@@ -15,6 +15,15 @@ import (
 // transactions that contain blobs.
 func (app *App) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
 	tx := req.Tx
+
+	// all txs should be under the max tx size limit
```

Comment:
why does this check need to be moved from the antehandler to check tx?

If we do this PR, do we still need the ante handler: max tx size?

Reply:
we can not change the AnteHandler in a patch because it'll be technically consensus breaking because it's called in process proposal. I have an issue open to clean up this logic + AnteHandlers in the next major release https://github.com/celestiaorg/celestia-app/issues/4087